pkg/virt-launcher/virtwrap/network/podinterface.go

```diff
@@ -490,7 +490,6 @@ func (b *BridgePodInterface) decorateConfig() error {
 	ifaces := b.domain.Spec.Devices.Interfaces
 	for i, iface := range ifaces {
 		if iface.Alias.Name == b.iface.Name {
-			ifaces[i].MTU = b.virtIface.MTU
```

Comment:
How would the vm nic get the correct MTU?
Are you planning to set the mtu on the tap device when creating it?
Why libvirt touch the mtu of an unmanaged device? Isn't it a libvirt bug?

Reply:
> How would the vm nic get the correct MTU?

It wouldn't.

> Are you planning to set the mtu on the tap device when creating it?

I don't think there is any alternative. The real issue is that netlink does not do that; as such, I'll have to patch netlink up, thus enabling it to set the MTU of tap interfaces. 

That is being worked on in [this PR](https://github.com/vishvananda/netlink/pull/583), which I haven't yet managed to get it to work.

> Why libvirt touch the mtu of an unmanaged device? Isn't it a libvirt bug?

Beats me. We say that we do not want libvirt to configure / manage the device, then proceed to explicitly say we want it to configure the MTU on it. It can be argued that our request is invalid - which I guess it would mean libvirt should refuse to handle the request.

@berrange what is your opinion about this ? Should we open a bug on libvirt ?

Reply:
AFAIR, libvirt only sets MTU on tap devices if you are telling libvirt to setup bridging (it copies MTU from the bridge to the TAP device), which you're not doing.

Reply:
@berrange I've successfully created a VMI w/ the following domxml interface snippet (using a pre-created tap device): 

```xml
    <interface type='ethernet'>
      <mac address='02:ca:fe:fa:ce:01'/>
      <target dev='tap0' managed='no'/>
      <model type='virtio'/>
      <mtu size='1000'/>
      <alias name='net0'/>
      <address type='pci' domain='0x0000' bus='0x01' slot='0x00' function='0x0'/>
    </interface>
```

The guest's MTU is 1000:

```bash
$ sudo virsh console myfc32
Connected to domain myfc32
Escape character is ^]

localhost login: root
Password: 
[systemd]
Failed Units: 1
  NetworkManager-wait-online.service
[root@localhost ~]# ip l
1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN mode DEFAULT group default qlen 1000
    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
2: eth0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1000 qdisc fq_codel state UP mode DEFAULT group default qlen 1000
    link/ether 02:ca:fe:fa:ce:01 brd ff:ff:ff:ff:ff:ff
    altname enp1s0
```

Reply:
Err, you have `<mtu size='1000'>`  which is explicitly telling libvirt you want it to set the MTU of the tap device.

Reply:
Right; that was my original question: if we say `managed='no'` but request an explicit MTU configuration then libvirt configures the MTU. Our assumption was that libvirt would not touch the interface (the managed='no' would take precedence).

@AlonaKaplan thinks this is a bug. I think this is a gray area ... that would definitely be clearer to the user if libvirt rejected this request. Hence my question to you.

Reply:
There's no bug here, just bad configuration. These are independent tunables and it is valid to say managed=no, and also ask for the MTU to be configured. If you don't want libvirt setting the MTU, then don't specify an MTU in the configuration,

Reply:
@berrange another question. If we don't pass the MTU of our unmanaged tap to libvirt, how would libvirt decide what MTU to pass to the qemu?

Reply:
If you don't specify MTU that libvirt won't do anything with MTU, nor will it tell QEMU to. If you need a specific MTU then set that up when creating the tap device in kubevirt - typically you need to know the host network connectivity to know what the right MTU to set is.

Reply:
@berrange if we set a specific MTU (e.g 8000) on the tap device (and don't pass any MTU to libvirt). How will the vm nic get this MTU (8000)?

Reply:
According to bug [1408701](https://bugzilla.redhat.com/show_bug.cgi?id=1408701) - "Libvirt automatically sets tap device's mtu to match the one that the bridge has (unless the tap device has different MTU set - in which case we also put it onto qemu's cmd line)."
Since our bridge has the correct MTU, I believe we should be fine.

Reply:
You're using type=ethernet, so libvirt will never look at/touch the bridge - it expects you to have setup the bridge device + tap device MTU beforehand to match.

The more important thing we're missing here is the passing of the "host_mtu" value to QEMU. I was under the impression that QEMU would query the TAP device MTU and pass that into the guest driver, but the code doesn't seem to actually do that. It appears as if host_mtu must always be set, and the only way to make that happen is to use the <mtu> element, but that in turn makes libvirt try to set the TAP device MTU which we don't want, to avoid permissions issues.